catalog/region-restriction-templates/RestrictToSpecifiedRegions.yaml

```
@@ -56,3 +62,22 @@
       %{ endfor }
   resources:
     - "*"
+
+- sid: "RestrictLambdaToSpecifiedRegions"
+  effect: "Deny"
+  actions:
+    - "lambda:*"
+  condition:
+    - test: "StringNotEqualsIgnoreCase"
+      variable: "aws:RequestedRegion"
+      # List of allowed regions
+      values:
+        # Us east-1 required for cloudfront lambda on edge
+      %{ if cloudfront_lambda_edge_enabled == "true" && !contains(split(",", allowed_regions), "us-east-1") }
+        - us-east-1
+      %{ endif }
+      %{ for r in split(",", allowed_regions) }
+        - ${trimspace(r)}
+      %{ endfor }
+  resources:
+    - "*"
```

Same idea as with `DenyLambdaInRegions`

```suggestion
%{ if cloudfront_lambda_edge_enabled && !strcontains(allowed_regions, "us-east-1") }
- sid: "RestrictLambdaToSpecifiedRegions"
  effect: "Deny"
  actions:
    - "lambda:*"
  condition:
    - test: "StringNotEqualsIgnoreCase"
      variable: "aws:RequestedRegion"
      # List of allowed regions
      values:
        # Us east-1 required for cloudfront lambda on edge
        - us-east-1
      %{~ for r in split(",", allowed_regions) ~}
        - ${trimspace(r)}
      %{~ endfor ~}
  resources:
    - "*"
%{ endif }
```